### ISSUE TITLE
Error when generate html doc

### ISSUE THREAD

**Issue by codex-corp** [jimschubert/intellij-swagger-codegen#4]
Can you provide quick note how to generate doc?

```
missing swagger input or config!
java.lang.RuntimeException: missing swagger input or config!
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:89)
	at com.jimschubert.intellij.swaggercodegen.ui.GenerateDialog.doOKAction(GenerateDialog.kt:266)
	at com.intellij.openapi.ui.DialogWrapper$OkAction.doAction(DialogWrapper.java:1871)
	at com.intellij.openapi.ui.DialogWrapper$DialogWrapperAction.actionPerformed(DialogWrapper.java:1837)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:675)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
	at java.awt.Dialog.show(Dialog.java:1084)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:779)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:457)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1693)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1635)
	at com.jimschubert.intellij.swaggercodegen.actions.CodegenGenerateAction.actionPerformed(CodegenGenerateAction.kt:30)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:199)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.a(ActionMenuItem.java:315)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:920)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:130)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:295)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.a(ActionMenuItem.java:112)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:91)
	at com.intellij.openapi.application.TransactionGuardImpl.access$300(TransactionGuardImpl.java:41)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:116)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:126)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:112)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.a(BegMenuItemUI.java:513)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:45)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:533)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:675)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by jimschubert** [comment]
This should work. Can you provide some repro steps so I can test this locally? It could be a change related to the recent update to swagger-codegen 2.2.0.

**Comment by codex-corp** [comment]
I just trying to generate simple yml file .. maybe I missed some
requirements in settings ?

**Comment by jimschubert** [comment]
I've just tried this against the `petstore.yaml` file from swagger-codegen's test resources, and generation works.

The steps are below. You can skip the parts for pulling swagger-codegen code if you already have a project and a yaml file open. I just wanted to include everything for completeness.

* git clone git@github.com:swagger-api/swagger-codegen.git
* open IntelliJ and go to File -> Open, browse to `swagger-codegen`
* In IntelliJ, open `modules/swagger-codegen/src/test/resources/2_0/petstore.yaml`
* (with the yaml file active) Go to Code -> Swagger -> Generate (⌥⇧G)
* Collapse the CLIENT and SERVER nodes, make sure DOCUMENTATION is expanded
* Select html in the left panel
* Set an output directory on the bottom of the dialog
* Click "Generate"

<img width="830" alt="generate_code_from_swagger_definition_and_petstore_yaml_-_swagger-codegen_-____projects_swagger-codegen_" src="https://cloud.githubusercontent.com/assets/109659/23006236/9ed09370-f3ce-11e6-8b01-871c37de3944.png">

There are some issues with usability that I plan to address when I get a chance. The biggest one being that there's not feedback that generation has completed. The error you posted suggests that the `Generate` option was available to you without an active document. I'd have to investigate that if this is the case.

If you don't necessarily need to be generating your HTML from within IntelliJ, you can generate it from the command line.

If you have Java installed, you can use [this script](https://gist.github.com/jimschubert/62baf653a9520b23399a67755c33631a) to download the swagger-codegen-cli jar directly from Maven and generate using the standard CLI options.

For example:

```bash
curl https://gist.githubusercontent.com/jimschubert/62baf653a9520b23399a67755c33631a/raw/c1a0e8e1b2d68fcac96c87e17027d45b8c02a51d/swagger.sh > swagger.sh
chmod u+x swagger.sh
./swagger.sh generate -l html -i http://petstore.swagger.io/v2/swagger.json -o docs
``` 

If you need to automate this in CI or something, you can use my docker image: https://hub.docker.com/r/jimschubert/swagger-codegen-cli/.

**Comment by jimschubert** [comment]
I've published version 0.0.4 of the plugin, which upgrades swagger-codegen. Please try that. I'm closing this issue as #5 should address the only other issue I found while generating docs.

**Comment by jimschubert** [comment]
@codex-corp I just published 0.0.5, which I intended only to show a "Generation succeeded" bubble when the generator succeeds and closes the dialog. I was able to repro the error you mentioned above by opening a swagger definition, opening the Swagger Codegen dialog and specifying an invalid path without selecting a language. There may be some other flow to get to this point, but it should be resolved now.

Sorry I wasn't able to repro before.
